### PR TITLE
Fix 4488 Multiple issues after rebases related to templates 

### LINKED ISSUE #4488: Multiple issues after rebases related to templates
Like in #4469 there are many places after rebase that requires some work. 
I have already did some work and will push PR after finish.

### PR DESCRIPTION
**Pull request by Matslom**
Resolves #4488
I've clicked across 1.9 and fixes some minor problems: uninitialized variables, unnecessary old template calls introduced in rebase, sql errors, some fixes in templates

### CODE REVIEW

**Comment by lairdshaw** [comment]
I've only just briefly visually inspected these changes, but want to thank you prior to a closer look for performing due post-rebase diligence. Much appreciated, @Matslom.

File: inc/views/base/forumdisplay/forumdisplay.twig [hunk 1]
```diff
@@ -231,48 +231,50 @@
...
-                        {{ include('partials/icon.twig', {icon: 'sort', class: 'select-field__icon'}, with_context = false) }}
-                        <select name="datecut">
-                            <option value="1"{% if mybb.input.datecut == 1 %} selected{% endif %}>{{ lang.datelimit_1day }}</option>
-                            {% for field in [5, 10, 20, 50, 75, 100] %}
-                                <option value="{{ field }}"{% if mybb.input.datecut == field %} selected{% endif %}>{{ attribute(lang, 'datelimit_' ~ field ~ 'days') }}</option>
-                            {% endfor %}
-                            <option value="365"{% if mybb.input.datecut == 365 %} selected{% endif %}>{{ lang.datelimit_lastyear }}</option>
-                            <option value="9999"{% if mybb.input.datecut == 9999 or mybb.input.datecut == '' %} selected{% endif %}>{{ lang.datelimit_beginning }}</option>
-                        </select>
+            {% if threadcount > 0 %}
```
Comment (Matslom) [inline inc/views/base/forumdisplay/forumdisplay.twig]: I've just found this #4479, so that block has to be reverted

**Review by cronlabspl — APPROVED** [review]
Looks good

**Comment by lairdshaw** [comment]
The TL;DR: LGTM but let's give it a day or so and then in the absence of objections merge this contribution.

This PR covers a lot, and although it's difficult to carefully review it given that it's not always clear what problem(s) it's trying to solve, I've tried to lightly test it under normal circumstances as well as to give the code itself a bit of a visual review, and it's turned out OK so far from my perspective. Because others might bring different perspectives, I won't merge it right away, but also because it's been such a long time since it was posted, I'll also only give it, say, a day or two more for anyone to object to its being merged.

**Comment by lairdshaw** [comment]
Merging in the absence of objections...
